### PR TITLE
Separate client code and api code

### PR DESCRIPTION
**Pull request by gnawux**
Under development, lack of login/logout, push/pull

Ready to review

Signed-off-by: Xu Wang <gnawux@gmail.com>

### CODE REVIEW

File: client/api/hijack.go [hunk 1]
```diff
@@ -1,24 +1,64 @@
...
+	return nil
+}
+
+func (cli *Client) dial() (net.Conn, error) {
 	return net.Dial(cli.proto, cli.addr)
 }
 
-func (cli *HyperClient) hijack(method, path string, setRawTerminal bool, in io.ReadCloser, stdout, stderr io.Writer, started chan io.Closer, data interface{}, hostname string) error {
+func (cli *Client) hijack(method, path string, setRawTerminal bool, in io.ReadCloser, stdout, stderr io.Writer, started chan io.Closer, data interface{}, hostname string) error {
 	defer func() {
```
Comment (laijs) [inline client/api/hijack.go]: setRawTerminal  is not a part of the api. it needs to be handle outside of the hijack()

**Comment by gnawux** [comment]
## Note

### Changed behavior

This patch only modified a small set of behavior

- support send signal to container with `kill` command
- deal with log from tty on non-tty container (together with #262 )

And for most of the command line, it behave as before. I do not want to change too much in one PR (it is already big enough).

### Refactor

move API related actions to `client/api` package, and leave the following in `client` package

- command line flags process
- output format
- retry and some plumbing jobs
- raw terminal related

**Comment by gnawux** [comment]
@laijs reviews are welcome

**Comment by feiskyer** [comment]
LGTM.

**Pull request merged**